common/lib/util/eventemitter.js

```diff
@@ -41,7 +41,7 @@ var EventEmitter = (function() {
 	 *        supplied, all listeners are removed.
 	 */
 	EventEmitter.prototype.off = function(event, listener) {
-		if(arguments.length == 0) {
+		if(arguments.length == 0 || (!event && !listener)) {
```

Comment:
I think this change means https://github.com/ably/ably-js/blob/e4c8cf7/common/lib/util/eventemitter.js#L69-L72 should now be unreachable, so can be removed